docs/en/operations/backup.md

```diff
@@ -356,7 +356,7 @@ RESTORE TABLE data AS data_restored FROM Disk('s3_plain', 'cloud_backup');
 :::note
 But keep in mind that:
 - This disk should not be used for `MergeTree` itself, only for `BACKUP`/`RESTORE`
-- It has excessive API calls
+- If your tables are backed by S3 storage, it doesn't use `CopyObject` calls to copy parts to the destination bucket, instead it downloads and uploads them, which is very inefficient. Prefer to use `BACKUP ... TO S3(<endpoint>)` syntax for this usecase.
```

Comment:
I guess by `BACKUP TO Disk('s3_backup')` you mean backup from `s3` to `s3_plain` disk.

And I think instead of adjusting documentation we can simply make `CopyObject` possible for this case, all you need to do is adjust `DataSourceDescription::sameKind` to consider `s3` and `s3_plain` as equal, this looks safe.

Reply:
> I guess by BACKUP TO Disk('s3_backup') you mean backup from s3 to s3_plain disk.

Yes, sorry, that's what I meant.

If it is safe it would be great, I can look into it. I still think it is worth to be explicit about the current state in the meanwhile, though.

Reply:
> If it is safe it would be great

It should, but it should be verified of course.

> I still think it is worth to be explicit about the current state in the meanwhile, though.

I don't have objections, documentation can be a good start anyway.